### PR TITLE
[agent-e][issue #2010] Add canonical author activity story owner

### PR DESCRIPTION
**Pull request by yazzaoui**
## Summary

- add the selected-store `author_activity_order` and `author_activity_occurrences` canonical owner for immutable author-facing occurrence identity and commit-visible cross-source order
- add the closed typed `AuthorActivity` registry, fail-closed transactional draft protocol, cursor reader, and safe human/plain/NDJSON renderers
- move every admitted producer family into the outer transaction on SQLite and PostgreSQL, with nested producers joining that owner and raw-transaction bypasses rejected
- add derived platform-event/effect disposition guards, backend parity, ordering, rollback, replay, conflict, privacy, and rendering proof

## Governing Contract

Authoritative sections changed in this PR:

- `platform-spec.yaml#platform_tables.author_activity_story`
- `platform-spec.yaml#platform_tables.tables.author_activity_order`
- `platform-spec.yaml#platform_tables.tables.author_activity_occurrences`
- `platform-spec.yaml#platform_tables.destructive_reset`

Binding gate context:

- [Pre-Implementation Coverage Audit](https://github.com/division-sh/swarm/issues/2010#issuecomment-4970918395)
- [independent gate outcome and conditions](https://github.com/division-sh/swarm/issues/2010#issuecomment-4971020919)

## Semantic Migration

**New canonical owner:** selected-store `author_activity_order` plus `author_activity_occurrences`, consumed through `internal/runtime/authoractivity`.

**Old paths now invalid:** post-commit story appends; timestamp, UUID, BIGSERIAL, or allocation-order cursors; current-state/log reconstruction; same-concept per-surface readers; registered producers operating through raw transactions; and nested producers opening a second transaction.

**Production paths checked:** inbound/events, entity mutations, delivery/dead letters, activity, effects, turns/tool completion, cards, agent lifecycle, directives, run lifecycle/fork/control/quiescence, and admitted platform signals. Runtime diagnostic logs, receipt evidence, migration/cleanup, and administrative settlement were independently classified as different concepts or represented by their canonical transition.

The migration is complete for the approved S1 failure class. Transport/feed attachment remains the explicitly approved S2 child of #2010; safe content summaries remain #2012 and liveness/detail facts remain #1564/#2031.

## Validation

- `go run ./cmd/swarm-test-postgres -- go test -p 1 ./... -count=1 -timeout=20m`
- focused SQLite/PostgreSQL semantic proof for serialization, parity, rollback, replay, conflicts, nested ownership, raw-writer rejection, registry coverage, privacy, and rendering

Part of #2010

### CODE REVIEW

**Comment by yazzaoui** [comment]
## Post-Implementation Proof Audit (revised for aa9f7b05)

### Scope and closure

- **Issue category:** high-risk semantic-drift / failure-class architecture work.
- **Exact concepts changed:** immutable author-facing occurrence identity; commit-visible order across heterogeneous runtime producers; atomic source-transition/story registration; closed per-kind source/projection/failure/subject semantics; stable cursor reads; and typed human/plain/NDJSON rendering.
- **Chosen working failure class closed entirely:** canonical_ordered_author_activity_story_owner_missing.
- **Immediate parent class:** supported author-activity surfaces do not yet consume one coherent live story.
- **Broader parent class:** operator-facing runtime observability remains fragmented across producer-specific facts, transports, and presentation dialects.
- **Issue framing:** #2010 is broader than this PR. The approved S1 boundary is a coherent first slice: this PR eliminates the durable owner/order/registry failure class across every admitted producer family. S2 remains the transport/feed consumer.
- **Achieved closure level:** **failure class eliminated for S1**.
- **Parent after this PR:** open. #2010 S2 must attach the serve feed; #2012 owns author-safe summaries; #1564/#2031 own liveness/detail facts.
- **Remaining child tail:** S2 replay/dedup/feed attachment is medium effort, high confidence. Summary and liveness fact owners remain separate. Review did not expand that tail.
- **Symptom-shaped issue:** yes at the parent level (silent serve output), but the approved S1 implementation boundary is owner-shaped rather than endpoint-shaped.

### Canonical owners after the change

1. **Normative owner:** platform-spec.yaml platform_tables.author_activity_story and its authoritative author_activity_order / author_activity_occurrences contracts.
2. **Commit/order owner:** internal/runtime/authoractivity.Mutation. It acquires the selected-store singleton before domain work, owns the exact outer transaction draft, assigns commit-visible sequence ranges, validates replay identity, and atomically finalizes source plus story.
3. **Per-kind semantic owner:** the single kindContracts table in internal/runtime/authoractivity. Each row owns admitted transitions, exact source owner, required source identity, allowed and required projection fields, failure-bearing transitions, subject strategy, and typed subject vocabulary.
4. **Read/render owner:** internal/runtime/authoractivity.List and Render. Subject rendering consumes the same per-kind contract; human prose is not a machine API.
5. **Turn input owner:** the already-canonical #2030 safe turn projector; this PR introduces no second turn decoder.

These are semantic owners, not nearby helpers: selected-store order is shared cross-source state, and kindContracts is the only admission and subject-selection authority.

### Touched consumers and exhaustive systematic-consumption audit

| Producer or consumer family | Classification after aa9f7b05 | Exact result |
|---|---|---|
| Inbound receipt and authored event append, including fork replay | moved to canonical owner | outer event transactions register inbound.received/v1 or event.emitted/v1; emitted events persist exact event type plus producer type/id, never event UUID as author subject |
| Entity current-state transitions | moved to canonical owner | mutation log, pipeline SQLite mutation, tools, setup, and fork materialization join the selected outer story transaction |
| Agent delivery receipts | moved to canonical owner | PostgreSQL and SQLite source helpers return changed; only a real delivery transition writes receipt timestamps and drafts delivery.lifecycle/v1 |
| Node/system delivery lifecycle | moved to canonical owner | existing changed/persisted-occurrence discipline remains the canonical distinct subscriber path |
| Dead-letter lifecycle | moved to canonical owner | store and pipeline writers register dead_letter.recorded/v1 in the source transaction |
| Activity lifecycle | moved to canonical owner | activity journal and fork activity generation register from canonical attempts |
| External effects | moved to canonical owner | all live effects.Registrations tuples have dispositions; effect stories persist adapter/transport/authority facts, never attempt UUID as author subject |
| Turns and tool completion | moved to canonical owner | LLM stores, completion settlement, and tool completion consume #2030 safe projections |
| Decision cards | moved to canonical owner | lifecycle mutations execute inside the story owner |
| Agent lifecycle | moved to canonical owner | admitted transitions use typed agent subject rules and exact diagnostic routes |
| Directives | moved to canonical owner | directive source writes and story registration share one transaction |
| Runs, forks, activation, control, completion, quiescence | moved to canonical owner | all admitted run transitions register from their real source mutation |
| Platform signals | moved to canonical owner | disposition is checked against the embedded spec catalog |
| Cursor reader | already consumes canonical owner | reads author_activity_occurrences.sequence exclusively after cursor with typed filters |
| Human/plain/NDJSON renderers | already consume canonical owner | validate each occurrence against kindContracts; subject strategy comes from the same row |
| Runtime diagnostic logs | different semantic concept, with proof | diagnostic evidence does not own author-facing lifecycle occurrence identity/order |
| Receipt evidence without a delivery transition | different semantic concept, with proof | no-op/duplicate source receipt is story-silent; it cannot duplicate a lifecycle occurrence |
| Administrative run-stop settlement | different semantic concept, with proof | canonical run cancellation represents the author-facing occurrence |
| Schema/destructive cleanup | different semantic concept, with proof | maintenance has no runtime occurrence; reset deletes occurrences and preserves singleton semantics |
| serve live feed | split / escalated as separate class | approved #2010 S2 consumes this owner |
| Author-safe summaries | split / escalated as separate class | #2012 |
| Agent liveness/detail | split / escalated as separate class | #1564/#2031 |
| Run-start follow | split / escalated as separate class | tracked run-follow work must consume this projection |

### Old paths now invalid or surviving

- **Invalid and guarded:** post-commit story append; timestamp/UUID/BIGSERIAL/allocation-order cursor; current-state/log reconstruction; raw transaction entry for registered producers; nested second transaction; ON CONFLICT DO NOTHING replay masking; producer-local subject precedence; event UUID or effect-attempt UUID as author subject; generic payload/evidence persistence; and drafting on a no-op mutable source write.
- **Removed in this revision:** separate transition/projection/failure switches; generic SubjectID-first rendering; unconditional receipt timestamp rewrite/story draft; obsolete SQLite terminal receipt helper.
- **Still authoritative:** runtime source tables own execution outcomes. Source UUIDs remain source evidence, not author-facing subjects.
- **Surviving same-concept bypass:** none found in the SQL writer, event catalog, effect registration, pipeline transaction, rendering, raw SQL, and selected-store sweeps.

### Manifestation coverage

| Known manifestation | Status | Exact proof |
|---|---|---|
| No durable owner for cross-source occurrence identity/order | reproduced and fixed | TestSQLitePostgresRegistryPersistenceParity |
| PostgreSQL allocation order can disagree with commit order | reproduced and fixed | TestPostgresSingletonOrdersByCommitAndRollbackReusesSequence and TestPostgresStore_RunEventTransactionSerializesStoryCommitOrder |
| Rollback can leak cursor gaps or story rows | reproduced and fixed | TestPostgresSingletonOrdersByCommitAndRollbackReusesSequence and TestSQLiteMutationOwnsRollbackReplayNestedBatchingAndPagination |
| Nested producers can escape atomicity | reproduced and fixed | TestSQLiteMutationOwnsRollbackReplayNestedBatchingAndPagination and existing nested-owner/raw-transaction rejection tests |
| Generic replay can hide conflicting story content | reproduced and fixed | TestSQLiteMutationRejectsConflictingReplayAtomically |
| Duplicate processed receipt can rewrite timestamps and conflict with the persisted delivered story | reproduced and fixed | TestAuthorActivityDuplicateTerminalReceiptIsNoOpParity runs the real UpsertEventReceipt path on SQLite and PostgreSQL, advances the SQLite clock, and proves unchanged source timestamps plus identical occurrence sets |
| SQLite/PostgreSQL occurrence and cursor semantics can diverge | reproduced and fixed | TestSQLitePostgresRegistryPersistenceParity |
| Registry can accept a wrong source owner or wrong typed subject | reproduced and fixed | TestKindContractsRejectWrongSourceOwnerAndSubjectType |
| Event adapter can persist/render raw event UUID instead of producer identity | reproduced and fixed | TestKindContractsRejectUnsafeEventAndEffectSubjectFields plus TestAuthorActivityEventAndEffectAdaptersRenderExactSubjects; exact line is “12:00:00  phrase-completer  emitted phrase.completed” |
| Effect adapter can persist/render raw attempt UUID instead of adapter metadata | reproduced and fixed | TestKindContractsRejectUnsafeEventAndEffectSubjectFields plus TestAuthorActivityEventAndEffectAdaptersRenderExactSubjects; exact line is “12:00:01  anthropic_api  in flight” |
| Event/effect subject rules can drift between producer and renderer | reproduced and fixed | TestRegistryAcceptsOnlyDeclaredKindsTransitionsAndProjectionFields and the exact adapter rendering test; both ValidateDraft and Render consume kindContracts |
| New platform event can bypass classification | reproduced and fixed | TestAuthorActivityPlatformEventDispositionCoversSpecCatalog |
| New effect adapter can inherit a guessed disposition | reproduced and fixed | TestAuthorActivityEffectDispositionCoversRegistrations |
| Unknown kind/transition/projection field or unsafe payload can enter the journal | reproduced and fixed | TestRegistryAcceptsOnlyDeclaredKindsTransitionsAndProjectionFields |
| Failure with poor optional detail can disappear | reproduced and fixed | TestAgentLifecycleFailureWithoutEnvelopeStillRendersDiagnosticRoute |
| Human rendering can become the machine API or leak ANSI/prose | reproduced and fixed | TestRenderModesKeepHumanProseSeparateFromTypedNDJSON |
| Raw SQL/direct selected-store writer can bypass the owner | reproduced and fixed | TestSelectedRawSQLBoundaryInventoryIsClassified, TestSelectedSQLiteRuntimeWriterBoundaryAudit, and the store abstraction guard matrix |
| Live serve feed is absent | split / escalated as separate class | #2010 S2 |
| Author-safe preview is absent | split / escalated as separate class | #2012 |
| Agent liveness/detail facts are absent | split / escalated as separate class | #1564/#2031 |

### Required proof run

- go run ./cmd/swarm-test-postgres -- go test ./internal/runtime/authoractivity ./internal/store -count=1 -timeout=15m — passed.
- go run ./cmd/swarm-test-postgres -- go test ./cmd/swarm -count=1 -timeout=15m — passed after removing two ignored stale tier-8 test databases created before the current master schema.
- go run ./cmd/swarm-test-postgres -- go test -p 1 ./... -count=1 -timeout=25m — clean complete repository suite passed.
- Focused semantic proof includes exact event/effect rendering and the real duplicate-receipt SQLite/PostgreSQL matrix.
- platform-spec.yaml is updated in this PR because this is platform architecture/runtime semantics work.
- S1 has no serve end-to-end claim; the supported serve feed proof belongs to approved S2.

### Sibling probe, watchlist, and architecture feedback

- **Sibling contexts checked:** event/inbound, entity, agent and node delivery, receipts, dead letters, activity, effects, turns/tools, cards, agent lifecycle, directives, runs/forks/control/quiescence, platform signals, logs, cleanup, renderers, raw SQL, and both selected-store backends.
- **Generic proof created:** one closed per-kind contract test family plus real adapter output and duplicate source-replay parity tests.
- **Watchlist decision:** no change. operator_surfaces.shared_observability_contract already captures raw identity/author-altitude drift and producer-coupled replay ordering; this PR removes those S1 manifestations.
- **Architecture smell:** source, projection, failure, and subject semantics were briefly split across parallel maps, producer choices, and generic renderer precedence. **Disposition: promoted and fixed in this PR** through kindContracts and changed-aware source helpers.
- **Tracking decision:** S1 architecture repair is complete here. Feed transport remains #2010 S2; summary/liveness remain #2012/#1564/#2031. No new issue or potential-issue entry is warranted.
- **Effort / ROI:** medium additional repair effort, high ROI. It prevents raw internal identities and duplicate receipt retries from corrupting the canonical story contract.
- **Optional non-closure-bearing feedback:** none.

File: internal/store/event_receipt_store.go [hunk 1]
```diff
@@ -419,30 +419,50 @@ func (s *PostgresStore) markEventDeliveryInProgressSpec(ctx context.Context, eve
...
+	transition := strings.TrimSpace(state.deliveryCode)
+	if transition == "" {
+		return fmt.Errorf("agent delivery author activity transition is required")
+	}
+	retry := state.retryCount
+	identity := delivery.deliveryID + ":" + transition + fmt.Sprintf(":%d", retry)
+	return runtimeauthoractivity.Record(ctx, runtimeauthoractivity.Draft{
+		Kind: runtimeauthoractivity.KindDeliveryLifecycle, Transition: transition,
+		SourceOwner: "event_deliveries", SourceIdentity: identity, DedupKey: "delivery:" + identity,
+		OccurredAt: occurredAt.UTC(), RunID: delivery.runID, EntityID: delivery.entityID, FlowID: delivery.flowInstance,
```
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/event_receipt_store.go:480]: Confirmed against current head `aca90e68`: this thread is still current and blocking. The implementation files are unchanged from the reviewed tree. Independent SQLite and PostgreSQL probes both fail a duplicate processed receipt with `author activity conflicting persisted replay for dedup key "delivery:<id>:delivered:0"`. The source helper should report no semantic state change and skip drafting on this replay (or otherwise reconstruct the exact persisted draft); add the duplicate-receipt proof on both selected stores.

File: internal/runtime/authoractivity/model.go [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+	KindCardLifecycle:      set("created", "decided", "deferred", "expired", "superseded"),
+	KindAgentLifecycle:     set("registered", "running", "terminated", "failed"),
+	KindDirectiveLifecycle: set("received", "in_flight", "completed", "failed", "outcome_uncertain"),
+	KindRunLifecycle:       set("started", "fork_prepared", "paused", "resumed", "fork_started", "completed", "failed", "cancelled", "forked"),
+	KindPlatformSignal:     set("agent_failed_retrying", "agent_failed", "event_quarantined", "dead_letters_escalated", "run_stalled", "runtime_reset", "authorization_required", "budget_warning", "budget_throttle", "budget_emergency", "budget_ok", "runtime_paused", "runtime_resumed", "recovery_failed"),
+}
+
+var allowedProjectionFields = map[Kind]map[string]struct{}{
+	KindInboundReceived:    set("subject_type", "subject_id", "provider"),
+	KindEventEmitted:       set("subject_type", "subject_id", "event_type", "producer_type", "producer_id"),
```
Comment (yazzaoui) [inline internal/runtime/authoractivity/model.go]: **[P1] Make the canonical registry own the author-safe subject shape**

The approved `event.emitted/v1` contract admits event type plus exact producer type/id, but this registry permits `subject_id` and does not require the producer. The live adapter therefore persists the event UUID as `SubjectID`, and `activitySubject` gives it precedence, rendering `event[<uuid>] emitted ...` instead of the author (the independent probe produced `event[event-uuid] emitted phrase.completed`). The same sibling seam exists for effects: `recordExternalEffectStory` persists the internal attempt UUID as the subject even though the approved fields are adapter/transport/authority metadata. This leaves author-altitude semantics split between adapters and a generic renderer and permits the raw-identity manifestation already named by the watchlist. Encode required/forbidden subject fields per kind in this canonical registry, remove internal event/attempt IDs from author-facing projection, make subject selection consume the approved producer/adapter/authority fields, and add counterexample renderer/registry tests for both kinds.

**Review by yazzaoui — COMMENTED** [review]
## Formal review: CHANGES NEEDED

### Blocking findings

1. **[P1] Duplicate terminal receipt replay is no longer idempotent.** `internal/store/event_receipt_store.go:191` and `internal/store/sqlite_runtime_delivery_replay.go:752-786` rebuild the stable `delivery:<delivery_id>:delivered:<retry>` occurrence with a fresh wall-clock timestamp even when the source delivery is already `delivered`. `Finalize` correctly rejects that different draft, so a normal duplicate `UpsertEventReceipt(..., processed, nil)` now returns `conflicting persisted replay` and rolls back. The existing inline thread remains unresolved and current. Independent real PostgreSQL and SQLite probes both reproduced the error. The approved protocol says replayed/no-op source operations append nothing; the clean fix is for the source transition helper to report whether state actually changed and draft only on change (or reconstruct the exact persisted source occurrence if the operation is genuinely a replay). Add the identical duplicate-receipt proof on both selected stores.

2. **[P1] The canonical registry does not own the required author-safe subject semantics.** `internal/runtime/authoractivity/model.go:162-176` only limits which field names may be non-empty; it does not require the approved fields or prevent an internal identity from becoming the human subject. Current producers exercise that gap: `internal/store/author_activity_events.go:83-84` persists the event UUID as `SubjectID` despite the approved `event.emitted/v1` shape being event type plus exact producer type/id, and `internal/store/runtime_external_effects.go:264-267` does the same with the internal attempt UUID despite the effect shape being adapter/transport/authority metadata. `internal/runtime/authoractivity/render.go:110-123` always gives `SubjectID` precedence. The independent renderer probe produced `event[event-uuid] emitted phrase.completed`, not `phrase-completer emitted phrase.completed`. This is the raw-identity/author-altitude manifestation already named by the watchlist. Encode each kind's required/forbidden subject shape in the registry, make subject selection consume that contract, and add event/effect counterexample tests. Do not patch this with renderer-only special cases while leaving invalid drafts admissible.

### Process and closure audit

The issue category is correctly high-risk semantic-drift/runtime architecture. The repaired Pre-Implementation Coverage Audit, independent Gate E `approved as first slice` plus broad-refactor approval, exact governing spec, and watchlist mapping all exist. Gate approval at 15:35 preceded implementation at 19:56. The authoritative `platform-spec.yaml#platform_tables.author_activity_story` is updated. The S1 class is a real coherent first slice, while S2 remains open in #2010.

The direction is good: the selected-store singleton, atomic source/story transaction, typed journal, cursor, and shared renderer are the correct bounded architecture. I found no alternate live story journal or same-concept reader outside this owner after sweeping event/inbound, entity, delivery/receipt/dead-letter, activity, effect, turn/tool, card, agent, directive, run/fork/control, platform signal, diagnostic log, raw event, trace, cleanup, and selected-store seams. The architecture smell is inside the new owner: exact projection semantics remain split across weak registry validation, producer-local field choices, and a generic subject fallback. **Disposition: promote now in this PR**, not a follow-up and not a broader framework.

The Post-Implementation Proof Audit is not merge-grade on current head. Its replay row cites only a synthetic conflicting-draft test and misses the live duplicate receipt producer; its privacy/rendering rows miss the event and effect raw-subject manifestations. Those claims are directly falsified above. Therefore `failure class eliminated` is not defensible; none of the merge closure levels is achieved until these same-class defects and proof-table rows are repaired.

### Independent proof and tracking

Passed:
- `go run ./cmd/swarm-test-postgres -- go test -p 1 ./internal/runtime/authoractivity ./internal/store ./internal/runtime/pipeline -run 'AuthorActivity|DeliveryOwnerInvariant|RunControl|RuntimeMutation|WorkflowInstanceStore_RunPipelineMutation' -count=1 -timeout=12m`
- Focused singleton-order, SQLite/PostgreSQL registry parity, platform/effect disposition, SQLite delivery, and PostgreSQL run-control tests under the PostgreSQL harness.
- `git diff --check`; all GitHub CI checks are green.

Failed as expected:
- Temporary backend-neutral authored-event render probe: got `event[event-uuid]`.
- Temporary duplicate processed-receipt probes on both SQLite and real PostgreSQL: got `conflicting persisted replay`.

Tracker state is independently verified: #2010, #1712, #1784, #1699, #1564, #2031, #2012, and #1820 are open; #2030 is closed. No new issue is warranted because both findings are closure-bearing S1 work. Watchlist decision: **no change**; `operator_surfaces.shared_observability_contract` already records raw identity/author-altitude drift and producer-coupled replay ordering. No `docs/POTENTIAL_ISSUES.md` exists and no potential-issue entry is needed.

### Reduce the next review cycle

Keep the current architecture. Make one table-driven registry contract per kind that covers allowed fields, required author-safe subject inputs, source owner/identity expectations, and failure requirements; have renderer subject selection consume that contract. Make mutable source helpers return `changed` plus persisted occurrence evidence and record only when `changed`. Then add one manifestation-level test per Post-Implementation Proof Audit row, including duplicate no-op replay and exact human subject output on both event and effect rows, and update the audit with those exact test names.

**Merge recommendation: changes needed; do not merge current head.**

**Comment by yazzaoui** [comment]
## Review checklist

- [x] PR description, full PR conversation, linked issue/thread, exact spec, implementer guidelines/checklist read
- [x] All inline threads checked; receipt replay thread is unresolved, current, and blocking
- [x] Category, pre-audit, independent Gate E timing, first-slice boundary, parent/sibling trackers, and watchlist independently verified
- [x] Authoritative `platform-spec.yaml` updated; zero legacy/migration seam retained
- [ ] Spec compliance: duplicate/no-op replay violates the atomic replay contract
- [ ] Failure-class coverage: receipt replay manifestation is live on PostgreSQL and SQLite
- [ ] Concept-wide drift reduction: event/effect author subjects still depend on producer-local raw IDs plus generic renderer precedence
- [ ] Post-Implementation Proof Audit: replay and privacy/render rows are under-enumerated and falsified
- **Risk level:** high
- **Closure level:** not proven; `failure class eliminated` rejected
- **Residual risk:** exact per-kind subject/source requirements are not enforced by the canonical registry
- **Watchlist:** no change; existing `shared_observability_contract` already captures both manifestations
- **Follow-up:** none; fix both blockers and their manifestation proofs in this PR
- **Merge recommendation:** **CHANGES NEEDED / DO NOT MERGE**

File: internal/store/event_receipt_store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (yazzaoui) [reply internal/store/event_receipt_store.go:480]: Fixed in aa9f7b05. Both receipt source helpers now return whether the canonical delivery tuple changed; no-op terminal replays neither rewrite delivery/receipt timestamps nor draft author activity. TestAuthorActivityDuplicateTerminalReceiptIsNoOpParity exercises the real duplicate processed-receipt path on SQLite and PostgreSQL and asserts unchanged timestamps plus identical occurrence sets. The clean full PostgreSQL-backed suite passes.

File: internal/runtime/authoractivity/model.go (hunk 2 above)
Comment (yazzaoui) [reply internal/runtime/authoractivity/model.go]: Fixed in aa9f7b05. A single kindContracts row now owns transitions, exact source owner/required identity, allowed and required projection fields, failure transitions, subject strategy, and typed subject vocabulary; both ValidateDraft and Render consume it. Event and effect producers no longer persist raw event/attempt IDs as subjects. The exact adapter test now renders phrase-completer emitted phrase.completed and anthropic_api in flight, with negative registry counterexamples for both raw-ID shapes.

**Comment by yazzaoui** [comment]
Review repair complete on aa9f7b05.

- Rebased onto current origin/master (3d5975d0); branch is exactly one commit ahead.
- Mutable PostgreSQL and SQLite receipt helpers now report changed and no-op duplicate terminal receipts neither rewrite source timestamps nor draft story rows.
- One per-kind kindContracts table now owns transitions, source requirements, allowed/required projection fields, failure transitions, and subject rendering.
- Raw event UUID and effect-attempt UUID subjects are removed; exact event/effect output and negative registry proofs are present.
- Added real duplicate processed-receipt parity proof on SQLite and PostgreSQL.
- Updated the authoritative platform-spec.yaml contract and replaced the Post-Implementation Proof Audit with exact manifestation rows.
- Clean full proof passed: go run ./cmd/swarm-test-postgres -- go test -p 1 ./... -count=1 -timeout=25m.

Both blocking inline threads have implementation replies and are resolved. Ready for re-review.

**Review by yazzaoui — COMMENTED** [review]
## Formal re-review: MERGE RECOMMENDED

Reviewed head `aa9f7b05ef5f965bb04adeaa243319169b5bdd8a` against base `3d5975d09679e49e829e6715e054fcd0488c51dc`.

### Findings

No blocking or non-blocking findings remain on the current head.

Both prior blockers are implementation-obsolete and their threads are resolved:

1. `internal/store/event_receipt_store.go:175-199,343-381` and `internal/store/sqlite_runtime_delivery_replay.go:737-799` now derive a source-level `changed` result, preserve source timestamps on an exact terminal replay, and draft author activity only for a real delivery transition. The selected-store parity test exercises the real duplicate processed-receipt path rather than a synthetic replay helper.
2. `internal/runtime/authoractivity/model.go:145-250,269-347` now has one per-kind contract row owning transitions, exact source owner, source-identity requirement, allowed and required projection fields, failure transitions, subject strategy, and typed subject vocabulary. `internal/runtime/authoractivity/render.go:122-150` consumes that same row. Event and effect producers no longer persist event/attempt UUIDs as author-facing subjects, and exact adapter output plus negative counterexamples cover both manifestations.

### Process and closure audit

The issue category remains correct: P1 high-risk semantic-drift, selected-store parity, runtime architecture, and operator-surface work. The repaired Pre-Implementation Coverage Audit and independent Gate E approval were recorded before implementation. The approved class is a coherent first slice of #2010, while S1 itself is the complete broad refactor required to eliminate `canonical_ordered_author_activity_story_owner_missing` across both selected stores and every admitted producer family.

The PR cites and updates the exact governing `platform-spec.yaml#platform_tables.author_activity_story` contract, including the mutable-source no-op rule and the single per-kind semantic owner. No legacy store, migration, backfill, fallback reader, timestamp cursor, dual owner, or compatibility seam remains.

The revised Post-Implementation Proof Audit is merge-grade. It names the canonical owners, classifies the repo-wide producer/consumer census, gives separate proof rows for all currently known S1 manifestations, states the S1/S2 boundary, and claims only `failure class eliminated for S1`. The manifestation table is now as extensive as the issue, gate, prior review, code sweep, and watchlist evidence support.

The issue framing is broader than this PR but acceptable as a first slice. No live same-concept author-story identity/order interpreter remains. Runtime logs, event follow, and trace remain diagnostic/raw concepts; #2030 remains the adjacent safe turn projector; S2 feed attachment and the summary/liveness consumers remain explicit separate classes. The patch materially reduces concept-wide drift by replacing producer-local order, admission, source, failure, and subject interpretation with one selected-store transaction owner and one closed kind contract.

### Independent adversarial verification

- Re-read the current PR body, all conversation, all reviews, and all review threads. Both prior threads are resolved and the old code anchors are obsolete on current head.
- Rechecked #2010, its repaired audit and Gate E record, the authoritative spec, and tracker state. #2010, #1712, #1784, #1699, #1564, #2031, #2012, and #1820 are open; #2030 is closed.
- Re-swept event/inbound, delivery/receipt/dead-letter, external effect, registry, renderer, pipeline, raw-writer, and selected-store seams. No alternate subject interpreter, replay writer, compatibility owner, or same-concept journal surfaced.
- Re-ran the original duplicate processed-receipt and event/effect raw-subject manifestations on real PostgreSQL and SQLite through their named tests; all pass.
- Added and removed an independent disposable probe for duplicate failed-terminal receipts. Both stores preserved source timestamps and the exact occurrence set.
- Verified all GitHub CI jobs are green and `git diff --check` passes.

### Architecture, tracking, and residual risk

The direction is good and bounded. The closure-bearing smell was split semantic ownership across parallel maps, producer-local subject choices, generic renderer precedence, and unconditional mutable-source drafting. **Disposition: promote now**, completed by this PR through `kindContracts` plus changed-aware source helpers. This is the minimum honest repair; generalized event sourcing or a compatibility layer would be over-engineering.

Watchlist decision: **no change**. `operator_surfaces.shared_observability_contract` already maps #2010 and records both raw-identity/author-altitude drift and producer-coupled commit-order/replay drift. No new potential-issue entry is warranted.

No new follow-up issue is needed for S1. Existing trackers already capture the remaining separate work: #2010 S2 for feed attachment, #1784/#1699 for run follow, #2012 for author-safe summaries, #1564/#2031 for liveness/detail facts, and #1820 as the broader detail-view parent.

Local proof passed:

- `go run ./cmd/swarm-test-postgres -- go test -p 1 ./internal/runtime/authoractivity ./internal/store -count=1 -timeout=15m`
- focused commit-order, registry, exact event/effect subject, and duplicate-receipt parity tests
- broader author-activity/store/pipeline transaction regression set
- independent duplicate terminal-failure replay probe on SQLite and PostgreSQL

Residual risk is the inherent breadth of the cross-source transaction migration, not a known uncovered manifestation. The complete focused packages, full repository CI, selected-store parity, and explicit S2 boundary reduce that risk to an acceptable merge level.

**Closure level: failure class eliminated for S1. Merge recommendation: merge current head.**

Implementer instruction: no further S1 code change is needed. Keep later S2 and consumer work on this cursor/registry owner; do not reintroduce runtime-log narration, timestamp replay, producer-local vocabulary, or compatibility behavior.

**Comment by yazzaoui** [comment]
## Re-review checklist (`aa9f7b05`)

- [x] PR description, full conversation, all reviews/threads, linked issue/thread, exact spec, guidelines, and checklist read
- [x] Both prior inline blockers verified obsolete and resolved on current head
- [x] Category, repaired pre-audit, independent Gate E timing/conditions, first-slice boundary, trackers, and watchlist independently verified
- [x] Authoritative `platform-spec.yaml` updated; zero legacy/migration/fallback seam retained
- [x] Spec compliance assessment passed
- [x] Failure-class coverage assessment passed for S1
- [x] Concept-wide drift-reduction assessment passed; no live same-concept interpreter found
- [x] Revised proof audit has canonical-owner accounting, repo-wide consumer accounting, manifestation-level proof, and explicit closure
- [x] Focused PostgreSQL/SQLite proof, broader transaction regression, full authoractivity/store tests, independent terminal-replay probe, `git diff --check`, and CI passed
- **Risk level:** high implementation scope; acceptable residual risk after proof
- **Closure level:** failure class eliminated for S1
- **Residual risk:** broad cross-source transaction migration only; no known unproven S1 manifestation
- **Watchlist:** no change; existing `shared_observability_contract` already captures the reusable class evidence
- **Follow-up:** already captured by #2010 S2, #1784/#1699, #2012, #1564/#2031, and #1820; no new issue
- **Merge recommendation:** **MERGE CURRENT HEAD**
